### PR TITLE
Add lzma encoding support via the 'xz' plugin

### PR DESCRIPTION
**Pull request by snorp**

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -114,6 +114,12 @@ adblock:
   optional: true
   listUrl: 'http://pgl.yoyo.org/adservers/serverlist.php?&mimetype=plaintext'
 
+# XZ compression settings.
+xz:
+  enabled: true
+  # Anything higher requires a lot of memory to decode
```
Comment (eamsen) [inline config/default.yml]: Missing `.`. Do we have a number for _a lot_ (just curious)?

File: config/test/test.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 
 cache:
   # make cache optionnal so we can disable it for testing
+  enabled: false
   optional: true
 
+logging:
+  level: error
```
Comment (eamsen) [inline config/test/test.yml:8]: Missing newline.

File: lib/cache.js [hunk 3]
```diff
@@ -29,15 +30,15 @@ var instance = null;
...
-  }
+  instance = new Cache(options);
 };
 
 // Saves new cache entry.
 exports.save = function(key, value, expire) {
   if (instance) {
     instance.save(key, value, expire);
+  } else {
+    log.warn('No cache instance');
```
Comment (eamsen) [inline lib/cache.js]: I'm thinking whether this should be an error. It's certainly something we would want to know in production.

File: lib/emit.js [hunk 4]
```diff
@@ -12,12 +12,12 @@ var Emitter = function(name) {
...
+Emitter.prototype.signal = function(type, value, arg) {
   var msg = this.prefix;
   if (value) {
     msg += '.' + value;
   }
-  this.emit(type, msg);
+  this.emit(type, msg, arg);
 };
 
 // Emitter index and factory.
```
Comment (eamsen) [inline lib/emit.js]: I'm having a deja vu here, didn't we merge this change some time ago?

File: lib/plugins/xz.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    'content-type': /(text\/|\/json|\/javascript|\/x-javascript)/,
+    'content-encoding': false
+  };
+
+  if (util.matchHeaders(request.headers, { 'accept-encoding': /xz/ }) &&
+      util.matchHeaders(source.headers, headerMatch)) {
+    emitter.signal('count', 'hit');
+    emitter.signal('start');
+
+    // We are writing gzip content
```
Comment (eamsen) [inline lib/plugins/xz.js]: `We are writing xz content.`

File: test/tests/global.js [hunk 6]
```diff
@@ -19,24 +19,28 @@ module.exports = {
...
       },
 
-      'simple no gzip': function(done) {
-        var req = helper.makeRequest(helper.getLocalUrl('dummy.html'),
-                                     '+gzip');
+    },
+
+    'xz': {
+
+      'simple accept xz': function(done) {
```
Comment (eamsen) [inline test/tests/global.js:35]: Maybe we should drop the `simple` as long as we don't have `complex`. Simple doesn't tell us much.

**Comment by eamsen** [comment]
With comments addressed, I think the code is good.
But we have a problem with the build times for lzma on Travis. Ideas?

File: config/default.yml (hunk 1 above)
Comment (snorp) [inline config/default.yml]: Level 8 requires 32MB to decode. Not sure what level 9 requires, but it's more than that, and I think 32MB per stream is plenty already.

File: lib/cache.js (hunk 3 above)
Comment (snorp) [inline lib/cache.js]: Fair enough. Changed to error.

File: lib/emit.js (hunk 4 above)
Comment (snorp) [inline lib/emit.js]: We did, it was based on that branch.

**Pull request merged**